File: AndroidAppBaseLibrary/src/main/java/stark/android/appbase/base/BaseLazyFragment.java
package stark.android.appbase.base;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;

/**
 * Created by jihongwen on 16/8/27.
 */

public abstract class BaseLazyFragment<T> extends BaseFragment<T> {

    protected boolean isVisible;

    protected boolean isViewCreated;

    protected boolean isLoaded;

    @Override
    public void setUserVisibleHint(boolean isVisibleToUser) {
        super.setUserVisibleHint(isVisibleToUser);
        if (getUserVisibleHint()) {
            isVisible = true;
            onVisible();
        } else {
            isVisible = false;
            onInvisible();
        }
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        isViewCreated = true;
    }

    protected void onVisible() {
        lazyLoad();
    }

    protected abstract void lazyLoad();

    protected void onInvisible() {

    }
}

File: AndroidAppBaseLibrary/src/main/java/stark/android/appbase/fragment/BaseLazyFragment.kt
package stark.android.appbase.fragment

import android.os.Bundle
import android.view.View

/**
 * Created by jihongwen on 16/8/27.
 */

abstract class BaseLazyFragment : BaseFragment() {

    private var isViewCreated: Boolean = false

    override fun setUserVisibleHint(isVisibleToUser: Boolean) {
        super.setUserVisibleHint(isVisibleToUser)
        if (userVisibleHint && isViewCreated) {
            lazyLoad()
        }
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        isViewCreated = true
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        lazyLoad()
    }

    protected abstract fun lazyLoad()

}
